base/exporter.go

```diff
@@ -0,0 +1,714 @@
+	ImportFeedMapStats *ExpVarMapWrapper `json:"import_feed"`
+}
+
+type ExpVarMapWrapper struct {
+	*expvar.Map
+}
+
+func (wrapper *ExpVarMapWrapper) MarshalJSON() ([]byte, error) {
+	return []byte(wrapper.String()), nil
+}
+
+type ReplicatorStats struct {
+	toplevelMap *expvar.Map
+}
+
+func (rs *ReplicatorStats) MarshalJSON() ([]byte, error) {
+	return []byte(rs.toplevelMap.String()), nil
+}
+
+func (rs *ReplicatorStats) Describe(ch chan<- *prometheus.Desc) {
+	return
+}
+
+func (rs *ReplicatorStats) Collect(ch chan<- prometheus.Metric) {
+	rs.toplevelMap.Do(func(value expvar.KeyValue) {
+		ch <- prometheus.MustNewConstMetric(
+			prometheus.NewDesc(prometheus.BuildFQName("sgw", "replication", "sgr_docs_checked_sent"), "sgr_docs_checked_sent", []string{"replication"}, nil),
+			prometheus.GaugeValue,
+			float64(value.Value.(*expvar.Map).Get("sgr_docs_checked_sent").(*expvar.Int).Value()),
+			value.Key,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			prometheus.NewDesc(prometheus.BuildFQName("sgw", "replication", "sgr_num_attachment_bytes_transferred"), "sgr_num_attachment_bytes_transferred", []string{"replication"}, nil),
+			prometheus.GaugeValue,
+			float64(value.Value.(*expvar.Map).Get("sgr_num_attachment_bytes_transferred").(*expvar.Int).Value()),
+			value.Key,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			prometheus.NewDesc(prometheus.BuildFQName("sgw", "replication", "sgr_num_attachments_transferred"), "sgr_num_attachments_transferred", []string{"replication"}, nil),
+			prometheus.GaugeValue,
+			float64(value.Value.(*expvar.Map).Get("sgr_num_attachments_transferred").(*expvar.Int).Value()),
+			value.Key,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			prometheus.NewDesc(prometheus.BuildFQName("sgw", "replication", "sgr_num_docs_failed_to_push"), "sgr_num_docs_failed_to_push", []string{"replication"}, nil),
+			prometheus.GaugeValue,
+			float64(value.Value.(*expvar.Map).Get("sgr_num_docs_failed_to_push").(*expvar.Int).Value()),
+			value.Key,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			prometheus.NewDesc(prometheus.BuildFQName("sgw", "replication", "sgr_num_docs_pushed"), "sgr_num_docs_pushed", []string{"replication"}, nil),
+			prometheus.GaugeValue,
+			float64(value.Value.(*expvar.Map).Get("sgr_num_docs_pushed").(*expvar.Int).Value()),
+			value.Key,
+		)
+	})
+}
+
+type DeltaSyncStats struct {
+	DeltaCacheHit             *SgwIntStat `json:"delta_cache_hit"`
+	DeltaCacheMiss            *SgwIntStat `json:"delta_cache_miss"`
+	DeltaPullReplicationCount *SgwIntStat `json:"delta_pull_replication_count"`
+	DeltaPushDocCount         *SgwIntStat `json:"delta_push_doc_count"`
+	DeltasRequested           *SgwIntStat `json:"deltas_requested"`
+	DeltasSent                *SgwIntStat `json:"deltas_sent"`
+}
+
+type QueryStats struct {
+	Stats map[string]*QueryStat
+	mutex sync.Mutex
+}
+
+type DbReplicatorStats struct {
+	NumAttachmentBytesPushed *SgwIntStat `json:"num_attachment_bytes_pushed"`
```

Comment:
Why the change in naming of these stats?

Reply:
I guess you're questioning why they're named differently in json rather than the exporter? If so there's no real reason. I'll change them to be the same.